app/src/ui/history/selected-commit.tsx

```diff
@@ -350,9 +350,40 @@ export class SelectedCommit extends React.Component<
         action: () => this.onOpenItem(file.path),
         enabled: isSafeExtension && fileExistsOnDisk,
       },
+      { type: 'separator' },
+      {
+        label: CopyFilePathLabel,
+        action: () => clipboard.writeText(fullPath),
+      },
     ]
+
+    let viewOnGitHubLabel = 'View on GitHub'
+    const gitHubRepository = this.props.repository.gitHubRepository
+
+    if (
+      gitHubRepository &&
+      gitHubRepository.endpoint !== getDotComAPIEndpoint()
+    ) {
+      viewOnGitHubLabel = 'View on GitHub Enterprise'
+    }
+
+    items.push({
+      label: viewOnGitHubLabel,
+      action: () => this.onViewOnGitHub(file),
+      enabled:
+        !this.props.isLocalRepository &&
```

Comment:
I assume this is supposed to mimic the `this.props.isLocal` check from https://github.com/desktop/desktop/blob/5d230b88021428960271180dc6ef392577fd204d/app/src/ui/history/commit-list-item.tsx#L371

However, that `this.props.isLocal` checks if the specific commit is local (i.e. not pushed) or not, since `View on GitHub` doesn't make sense for commits that only exist locally.

I just tested the current implementation and you can try and open in GitHub commit files that haven't been pushed:

![2021-07-26 at 18 20](https://user-images.githubusercontent.com/1083228/127023828-e6188878-b931-410e-81f5-f75948b2ba8c.png)

![2021-07-26 at 18 21](https://user-images.githubusercontent.com/1083228/127023895-ddf054bc-dd2a-4460-97ea-fc7d1002f200.png)

Reply:
Fixed in b054452d87a5fed833844015e682af2c29302589